src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -64,7 +64,7 @@ public async Task<bool> IsProjectDebuggableAsync(DebugLaunchOptions launchOption
             try
             {
                 // Launch providers to enforce requirements for debuggable projects
```

Comment:
>Launch providers to enforce requirements for debuggable projects [](start = 19, length = 64)

This comment is no longer true then?

Reply:
@ocallesp Will need to comment on that. I wasn't sure if launching the debug target was intended or not.

Reply:
Yes. It was intended.  The idea is to hit ProjectNotRunnableDirectlyException in one of the debug providers.

Reply:
Passing argument False will display class libraries in the startup. I need to reopen the issue.

Reply:
@ocallesp You can use it this way. Passing true means that the project is being told to debug the app. For webs this means starting web servers, launching browsers etc. For Functions projects, it downloads runtimes, starts the Azure storage emulator and runs the project.

Reply:
`IsProjectDebuggableAsync`() is only used in `StartupProjectRegistrar.cs`. My assumption was that `QueryDebugTargetsInternalAsync`() was only about to query, not execute.

Reply:
Comment removed.

Reply:
For simple projects like console applications it amounts to the same thing. But for more complication apps it doesn't. From a VS perspective the flag setting represents whether [IVsDebuggableProjectCfg](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivsdebuggableprojectcfg?view=visualstudiosdk-2019) QueryDebugLaunch() was called, DebugLaunch() was called.